cs/src/core/Index/Common/RecordInfo.cs

```diff
@@ -67,33 +68,29 @@ public static void WriteInfo(ref RecordInfo info, bool inNewVersion, bool tombst
             info.InNewVersion = inNewVersion;
         }
 
+        public bool IsLocked => (word & (kExclusiveLockBitMask | kSharedLockMaskInWord)) != 0;
+
+        public bool IsLockedExclusive => (word & kExclusiveLockBitMask) != 0;
+
+        public bool IsLockedShared => (word & kSharedLockMaskInWord) != 0;
+
+        public bool IsIntermediate => (word & (kStubBitMask | kSealedBitMask)) != 0;
+
         /// <summary>
         /// Take exclusive (write) lock on RecordInfo
         /// </summary>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void LockExclusive()
```

Comment:
isnt it more efficient to inline this lock case instead of reusing Try, which requires while, comparisons to spincount, etc.